### PR TITLE
Implement String.prototype.search, and some minor regexp refactors.

### PR DESCRIPTION
**Pull request by zherczeg**
JerryScript-DCO-1.0-Signed-off-by: Zoltan Herczeg zherczeg@inf.u-szeged.hu

### CODE REVIEW

**Comment by zherczeg** [comment]
Any thoughts?

File: jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp [hunk 1]
```diff
@@ -31,6 +31,10 @@
 #include "jrt-libc-includes.h"
 #include "lit-char-helpers.h"
 
+#ifndef CONFIG_ECMA_COMPACT_PROFILE_DISABLE_REGEXP_BUILTIN
+#include "ecma-regexp-object.h"
+#endif
```
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp]: Usually, `#endif` is followed by a condition, under which block above is executed.
In the case: `/* !CONFIG_ECMA_COMPACT_PROFILE_DISABLE_REGEXP_BUILTIN */`

File: jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp [hunk 2]
```diff
@@ -1517,11 +1509,91 @@ ecma_builtin_string_prototype_object_replace (ecma_value_t this_arg, /**< this a
...
+  }
+
+  ECMA_FINALIZE (to_string_value);
+  ECMA_FINALIZE (check_coercible_value);
+
+  /* 6. */
+  return ret_value;
 } /* ecma_builtin_string_prototype_object_search */
 
+#endif /* CONFIG_ECMA_COMPACT_PROFILE_DISABLE_REGEXP_BUILTIN */
```
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp]: Maybe, `/* !CONFIG_ECMA_COMPACT_PROFILE_DISABLE_REGEXP_BUILTIN */`

**Comment by ruben-ayrapetyan** [comment]
I've got no other comments.

**Comment by zherczeg** [comment]
Thank you. Your comments were fixed.

**Comment by kkristof** [comment]
Looks good to me.

**Comment by egavrin** [comment]
`make push`

**Comment by zherczeg** [comment]
Landed in 0a1b6eb4c7d432b47f95f1f248fde562889770df

**Pull request closed without merging**